### DISCUSSION TITLE
How can I specify the type of a Block particle?

### DISCUSSION THREAD

**Discussion by ryleu** [Minestom/Minestom#1146] [Help] [answered]
I'm trying to show the particle that appears when a player hits the ground hard, and I have this so far:

```kt
player.sendPacketToViewersAndSelf(
    ParticleCreator.createParticlePacket(
        Particle.BLOCK,
        false,
        pos.x, pos.y + 0.25, pos.z,
        1f, 0.25f, 1f,
        0f,
        30
    ) {}
)
```

How can I specify the type of block (like `minecraft:dirt`, for example) that the particle displays?

**Comment by Tim3Game** [comment, marked as answer]
Hey! Sorry it's not Kotlin, but I have a working example in Java.
```java
player.sendPacketToViewersAndSelf(
        ParticleCreator.createParticlePacket(
                Particle.BLOCK,
                false,
                pos.x(),
                pos.y() + 0.25,
                pos.z(),
                1f,
                0.25f,
                1f,
                0f,
                30,
                binaryWriter -> {
                    binaryWriter.writeVarInt(Block.STONE.stateId());
                }
        )
);
```
By this, you can probably guess how to have different block particles _(hint: you simply change the STONE with some other block)_. You can find on [https://wiki.vg/Protocol#Particle](https://wiki.vg/Protocol#Particle) what field type(s) to use in different particles requiring extra data. A help was issue #721.
/edit: Changed ``id()`` to ``stateId()`` to reflect ryleu's findings.

**Reply by ryleu** [reply]
Awesome, thank you!

Final Kotlin code:

```kt
player.sendPacketToViewersAndSelf(
    ParticleCreator.createParticlePacket(
        Particle.BLOCK,
        false,
        taskPos.x, taskPos.y + 0.2, taskPos.z,
        3f, 0f, 3f,
        0f,
        100
    ) {
        it.writeVarInt(Block.DIRT.id())
    }
)
```

**Reply by ryleu** [reply]
UPDATE: Turns out that you need to use `stateId()` instead of `id()`, or else it won't be correct for all blocks.
